src/datasets/streaming.py

```diff
@@ -112,3 +111,16 @@ def extend_dataset_builder_for_streaming(builder: "DatasetBuilder"):
                 internal_import_name = imports[1]
                 internal_module_name = ".".join(builder.__module__.split(".")[:-1] + [internal_import_name])
                 extend_module_for_streaming(internal_module_name, use_auth_token=builder.use_auth_token)
+
+    # builders can inherit from other builders that might use streaming functionality
+    # (for example, ImageFolder and AudioFolder inherit from FolderBuilder which implements examples generation)
+    # but these parents builders are not patched automatically as they are not instantiated, so we patch them here
+    from .builder import DatasetBuilder
+
+    parent_builder_modules = [
+        cls.__module__
+        for cls in type(builder).__mro__[1:]
+        if issubclass(cls, DatasetBuilder) and cls is not DatasetBuilder
```

Comment:
The windows CI fails when running the CSV builder, which is a GeneratorBasedBuilder. I took a look at the only changes that could affect a GeneratorBasedBuilder and I think this line caused the CI to hang on windows by patching `builder.py` for streaming and failing.

Maybe we can avoid patching `builder.py` for now ?
```suggestion
        if issubclass(cls, DatasetBuilder) and cls.__module__ != DatasetBuilder.__module__]
```